### DISCUSSION TITLE
\src\prompts\chat\config.json'. The file is either invalid or missing

### DISCUSSION THREAD

**Discussion by derhallim** [microsoft/teams-sdk#1464] [Q&A]
I've set up a new bot from scratch, using the basic bot with TypeScript. 

The folder structure I have looks like this: 

![image](https://github.com/microsoft/teams-ai/assets/7239963/7a828189-f1c6-4df9-8703-dbc415287879)

My config.json under prompts look like this: 
![image](https://github.com/microsoft/teams-ai/assets/7239963/54465b9a-9035-45d7-919d-015533bc6eaa)

and this is how my index.ts look like: 

![image](https://github.com/microsoft/teams-ai/assets/7239963/dad1a990-8691-42d7-a3be-34447b931c9b)

However, when I hit F5, and send the bot any message, I get the following error: 
![image](https://github.com/microsoft/teams-ai/assets/7239963/218bcfc7-1be2-4950-8276-3d65ea19e131)

Any idea where I should be looking at?

Thanks all for your help!

**Comment by derhallim** [comment]
@corinagum, do you happen to know the reason for this? Also, is this the right place to post such questions?

**Reply by corinagum** [reply]
This means that your config.json for your prompts is not created or set up incorrectly. Please take a look at light bot sample and see the prompts folder. Our getting started documentation will also have more information on prompts.

**Reply by derhallim** [reply]
Hi @corinagum , I think I'm doing it correctly, I actually copied the same config.json in the twenty questions sample and just modified the description, and now I copied the config.json from the light bot and modified the description and the model name, same problem: 

![image](https://github.com/microsoft/teams-ai/assets/7239963/2f75cbbf-1e7c-4628-b106-9052ae7483af)

All config.json examples I've used are from the sample bots in this repo so.. there must be something else that's causing it.

**Reply by corinagum** [reply]
@derhallim Would it be possible to share your code in a repo? Your `config.json`s don't appear to have errors, but whether it's correct for your app depends on what your app is doing. For example, augmentations need an actions.json file adjacent to the config.json.

Your path to the prompts folder in `PromptManager` seems correct, but just in case could you confirm you can console.log the contents of the `config.json` using that path to rule out the 'config file is missing' part of the error?

If you can't share your repo, please copy-paste the following:

1. stack trace of your error & logs that are being consoled from `logRequests = true`
2. code snippet of where the planner is called as well as the app routing to that call.
3. skprompt.txt content 
4. If you are using actions, please share your actions.json

Please make sure any personally identifiable information is removed before sharing.

**Reply by derhallim** [reply]
@corinagum  here's the repo: https://github.com/derhallim/AskJoe/tree/master

**Reply by corinagum** [reply]
1. Please see our migration docs. It looks like you haven't yet updated the adapter -- our samples show how `TeamsAdapter` is used.
2. You should delete line 62 -- it is unused and not compatible with our library.
3. Your code does not have any route handling such as `app.message` -- there's no coded behavior on how the bot should respond to a message.

Without seeing a stack trace, I would assume that a combination of the above is sending the code down an unknown path. I would start with the above as well as deleting the prompt manager code temporarily. Once you get an equivalent to echobot working, you'll be ready to add prompts back in and work up from there.

Again I highly recommend reading our getting started documentation - there are a lot of topics covered that will help with understanding the samples.

**Reply by derhallim** [reply]
Thanks @corinagum , will check it out and get back to you!
